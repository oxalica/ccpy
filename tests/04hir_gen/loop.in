def f():
  while True:
    a

def g():
  for x in xs:
    b
