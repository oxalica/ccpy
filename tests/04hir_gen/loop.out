Start(Local size = 3):
0: %0 = __intrinsic__tuple_make_()
1: %0 = Closure #1() defaults = %0
2: %1 = __intrinsic__get_global0()
3: %2 = "f"
4: %0 = __intrinsic__dict_set3(%1, %2, %0)
5: %0 = __intrinsic__tuple_make_()
6: %0 = Closure #2() defaults = %0
7: %1 = __intrinsic__get_global0()
8: %2 = "g"
9: %0 = __intrinsic__dict_set3(%1, %2, %0)


Closure #1(Local size = 5):
0: %0 = True
1: %1 = __intrinsic__get_global0()
2: %2 = "__builtin__not"
3: %3 = None
4: %1 = __intrinsic__dict_get3(%1, %2, %3)
5: %0 = __intrinsic__tuple_make_(%0)
6: %0 = __intrinsic__v_call2(%1, %0)
7: %1 = __intrinsic__get_global0()
8: %2 = "__builtin__not"
9: %3 = None
10: %1 = __intrinsic__dict_get3(%1, %2, %3)
11: %0 = __intrinsic__tuple_make_(%0)
12: %0 = __intrinsic__v_call2(%1, %0)
13: JmpIfFalse %0 <23>
14: %1 = "a"
15: %2 = __intrinsic__get_global0()
16: %3 = "__builtin__global_get"
17: %4 = None
18: %2 = __intrinsic__dict_get3(%2, %3, %4)
19: %1 = __intrinsic__tuple_make_(%1)
20: %1 = __intrinsic__v_call2(%2, %1)
21: %1 = False
22: JmpIfFalse %1 <0>
23: %0 = None
24: Return %0


Closure #2(Local size = 8):
0: %1 = "xs"
1: %2 = __intrinsic__get_global0()
2: %3 = "__builtin__global_get"
3: %4 = None
4: %2 = __intrinsic__dict_get3(%2, %3, %4)
5: %1 = __intrinsic__tuple_make_(%1)
6: %1 = __intrinsic__v_call2(%2, %1)
7: %2 = __intrinsic__get_global0()
8: %3 = "__builtin__iter"
9: %4 = None
10: %2 = __intrinsic__dict_get3(%2, %3, %4)
11: %1 = __intrinsic__tuple_make_(%1)
12: %1 = __intrinsic__v_call2(%2, %1)
13: PushExcept %3 <30>
14: %2 = __intrinsic__get_global0()
15: %3 = "__builtin__next"
16: %4 = None
17: %2 = __intrinsic__dict_get3(%2, %3, %4)
18: %3 = __intrinsic__tuple_make_(%1)
19: %2 = __intrinsic__v_call2(%2, %3)
20: %0 = %2
21: %3 = "b"
22: %4 = __intrinsic__get_global0()
23: %5 = "__builtin__global_get"
24: %6 = None
25: %4 = __intrinsic__dict_get3(%4, %5, %6)
26: %3 = __intrinsic__tuple_make_(%3)
27: %3 = __intrinsic__v_call2(%4, %3)
28: %3 = False
29: JmpIfFalse %3 <14>
30: %4 = "StopIteration"
31: %5 = __intrinsic__get_global0()
32: %6 = "__builtin__global_get"
33: %7 = None
34: %5 = __intrinsic__dict_get3(%5, %6, %7)
35: %4 = __intrinsic__tuple_make_(%4)
36: %4 = __intrinsic__v_call2(%5, %4)
37: %5 = __intrinsic__get_global0()
38: %6 = "__builtin__isinstance"
39: %7 = None
40: %5 = __intrinsic__dict_get3(%5, %6, %7)
41: %4 = __intrinsic__tuple_make_(%3, %4)
42: %4 = __intrinsic__v_call2(%5, %4)
43: JmpIfFalse %4 <46>
44: %5 = False
45: JmpIfFalse %5 <47>
46: Raise %3
47: %1 = None
48: Return %1


